pkg/server/commands/write_authzmodel_test.go

```diff
@@ -0,0 +1,70 @@
+package commands
+
+import (
+	"context"
+	"testing"
+
+	"github.com/oklog/ulid/v2"
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	mockstorage "github.com/openfga/openfga/internal/mocks"
+	"github.com/openfga/openfga/pkg/typesystem"
+)
+
+func TestWriteAuthorizationModelWithExperimentalEnableModularModels(t *testing.T) {
+	ctx := context.Background()
+	storeID := ulid.Make().String()
+
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+
+	mockDatastore := mockstorage.NewMockOpenFGADatastore(mockController)
+	mockDatastore.EXPECT().MaxTypesPerAuthorizationModel().AnyTimes().Return(100)
+
+	testCases := map[string]struct {
+		enableModularModules bool
+		inputSchema          string
+		expectAllowed        bool
+	}{
+		`allow`: {
+			enableModularModules: true,
+			inputSchema:          typesystem.SchemaVersion1_2,
+			expectAllowed:        true,
+		},
+		`forbid`: {
+			enableModularModules: false,
+			inputSchema:          typesystem.SchemaVersion1_2,
+			expectAllowed:        false,
+		},
+	}
```

Comment:
will it make sense to add test to ensure schema 1.1 is still allowed if enableModularModules = false?

Reply:
Yeah that makes sense, will add

Reply:
I think this is covered by the existing tests. We should add one that ensures 1.1 is accepted if enableModularModules = true

Reply:
Tests added in 3d14bf43dc9e2d0b6ad2a4f93503c0d4cf2f0de6